### ISSUE TITLE
Code Issue

### ISSUE THREAD

**Issue by Keep-lucky** [ZubinGou/multi-view-prompting#4]
I would like to perform fine-tuning training on a custom dataset. Could you please let me know which parameters and files I need to modify? What does the 'const.py' file represent? Do I need to build it based on my custom dataset?

**Comment by ZubinGou** [comment]
You can modify only the data and configure the data path. You may want to calculate the optimized orders with [this code](https://github.com/ZubinGou/multi-view-prompting/blob/d91f3ec49d09a31ec5eada8c98b3a2c53b54f23e/src/data_utils.py#L28) and put the orders in `const.py`.

1. If it is different from any dataset in our code base, you need to build a new category list (such as [rest_cate_list](https://github.com/ZubinGou/multi-view-prompting/blob/d91f3ec49d09a31ec5eada8c98b3a2c53b54f23e/src/const.py#L6C7-L6C7)) and calculate the optimal orders of elements required in your custom dataset (such as [O][A][C][S]) via the function [choose_best_order_global](https://github.com/ZubinGou/multi-view-prompting/blob/d91f3ec49d09a31ec5eada8c98b3a2c53b54f23e/src/data_utils.py#L167C13-L167C13), then update the dict [optim_orders_all](https://github.com/ZubinGou/multi-view-prompting/blob/d91f3ec49d09a31ec5eada8c98b3a2c53b54f23e/src/const.py#L104C1-L104C17).

2. You need to update the file 'force_tokens.json' for constrained decoding. Refer to [prefix_allowed_tokens_fn](https://github.com/ZubinGou/multi-view-prompting/blob/d91f3ec49d09a31ec5eada8c98b3a2c53b54f23e/src/main.py#L321). Note that you should first modify the category list of your dataset before doing this.

**Comment by ZubinGou** [comment]
The categories to which the aspect terms belong depend on your dataset. You only need to build the category list if you want to use constrained decoding, which is optional.

**Comment by Keep-lucky** [comment]
> The categories to which the aspect terms belong depend on your dataset. You only need to build the category list if you want to use constrained decoding, which is optional.

Thank you very much for your response. I wish you success in your work and a delightful life!

**Comment by Keep-lucky** [comment]
1.Following your advice, I have been using constrained decoding on a custom dataset. While constructing the 'force_tokens.json' file, I noticed that the prefix_allowed_tokens_fn function may require some additional information. Upon comparing it with the original code in 'force_tokens.json', I believe it would be appropriate to make the following modifications by adding a dictionary for each task name.
https://github.com/ZubinGou/multi-view-prompting/blob/d91f3ec49d09a31ec5eada8c98b3a2c53b54f23e/src/main.py#L343-L345
![556b170b92a93b87569557da2749412](https://github.com/ZubinGou/multi-view-prompting/assets/71928923/02c3afd2-4b01-40f5-9b2b-a28ec4e9abea)
![image](https://github.com/ZubinGou/multi-view-prompting/assets/71928923/62540e61-aa3a-4e80-9c24-d59903b125e8)

2.Additionally, I have encountered some issues during the decoding process in inference. Specifically, when decoding sentences consisting of only two words, such as "Great price." the output is limited to [O] great [A] price, omitting the sentiment [S] positive tags. I was wondering if you came across a similar problem during your experimentation. Could this issue be related to my utilization of constrained decoding or the [O] [A] [S] element order prompts?

I am eagerly awaiting your response!
